### PR TITLE
infoschema, executor: Add SQL text column to DATA_LOCK_WAITS table

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What problem does this PR solve?

Problem Summary: This PR adds `SQL_DIGEST_TEXT` column to `DATA_LOCK_WAITS` table so that the SQL text corresponding to the `SQL_DIGEST` column can be displayed.

### What is changed and how it works?

What's Changed:

* Adds `SQL_DIGEST_TEXT` column to `DATA_LOCK_WAITS` table that shows the SQL text corresponding to the `SQL_DIGEST` column.
* Moves tests for `DATA_LOCK_WAITS` to `testClusterTableSuite`, which involves some changes to the mockStorage

Requires https://github.com/pingcap/tidb/pull/26357

### Check List 

Tests 

- [x] Unit test

Side effects

- [x] Performance regression: Consumes more CPU
- [x] Performance regression: Consumes more Memory

Documentation

- [ ] Affects user behaviors

### Release note 

```release-note
Add SQL text information to `INFORMATION_SCHEMA.DATA_LOCK_WAITS` table
```

### CODE REVIEW

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -2256,6 +2214,133 @@ func (e *tidbTrxTableRetriever) retrieve(ctx context.Context, sctx sessionctx.Co
...
+			}
+		}
+
+		var digests []string
+		if needDigest || needSQLText {
+			digests = make([]string, end-start)
+			for i, lockWait := range r.lockWaits {
+				digest, err := resourcegrouptag.DecodeResourceGroupTag(lockWait.ResourceGroupTag)
+				if err != nil {
+					logutil.BgLogger().Warn("failed to decode resource group tag", zap.Error(err))
```
Comment (cfzjywxk) [inline executor/infoschema_reader.go:2269]: Seems there's input `ctx`, we could use `logger` with it. We could add some comments if the error is expected to be ignored.

File: executor/infoschema_reader.go [hunk 2]
```diff
@@ -2256,6 +2214,133 @@ func (e *tidbTrxTableRetriever) retrieve(ctx context.Context, sctx sessionctx.Co
...
+					digests[i] = hex.EncodeToString(digest)
+				}
+			}
+		}
+
+		// Fetch the SQL Texts of the digests above if necessary.
+		var sqlRetriever *SQLDigestTextRetriever
+		if needSQLText {
+			sqlRetriever = NewSQLDigestTextRetriever()
+			for _, digest := range digests {
```
Comment (youjiali1995) [inline executor/infoschema_reader.go:2280]: What if the `digest` is a emtpy string since you think it's possible that fails to `DecodeResourceGroupTag`?

Comment (MyonKeminta) [reply executor/infoschema_reader.go:2280]: Ah, I forgot to handle this case. But luckily it doesn't affect the final result.

**Review by youjiali1995 — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]
/lgtm

**Pull request merged**